### DISCUSSION TITLE
How do I force static linkage against usrsctp.

### DISCUSSION THREAD

**Discussion by Nemirtingas** [microsoft/vcpkg#20027] [Q&A] [answered]
Hi,

I'm compiling statically libdatachannel but the result shared library on Linux and MacOS is dynamically linked against usrsctp.so/.dylib. It seems the usrsctp CMake target builds static and dynamic library and they are both installed in the vcpkg output directory.
When I then build my project, it links against the dynamic library, if I remove all .so/.dylib, it links against the static library.

How do I force static linkage when the shared library is also present ?

**Comment by autoantwort** [comment, marked as answer]
Can you try #20028, then there should be only a static lib.

**Reply by Nemirtingas** [reply]
Hi.
Thanks I will try it. I'll report back ASAP.

**Reply by Nemirtingas** [reply]
It seems that there is only the static library now in the output folder. It compiled successfully but I did not test it runtime.

**Reply by Nemirtingas** [reply]
It works. Thanks.
